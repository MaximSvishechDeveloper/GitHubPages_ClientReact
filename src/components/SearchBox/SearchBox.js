import React, { useState, useEffect, useCallback } from "react";

const SearchBox = ({ searchProfile }) => {
  const [search, setSearch] = useState("");

  const onSubmit = useCallback(() => {
    searchProfile(search);
  }, [search, searchProfile]);

  useEffect(() => {
    const keyDownHandler = (event) => {
      if (event.key === "Enter") {
        event.preventDefault();

        onSubmit();
      }
    };

    document.addEventListener("keydown", keyDownHandler);

    return () => {
      document.removeEventListener("keydown", keyDownHandler);
    };
  }, [onSubmit]);

  return (
    <div className="pa2">
      <input
        className="pa3 ba b--green bg-lightest-blue turquoise"
        type="search"
        placeholder="Search GitHubPage"
        onChange={(event) => {
          setSearch(event.target.value);
        }}
      />
      <button
        className="pa3 ba b--green bg-lightest-blue turquoise"
        onClick={onSubmit}
      >
        Search
      </button>
    </div>
  );
};

export default SearchBox;
