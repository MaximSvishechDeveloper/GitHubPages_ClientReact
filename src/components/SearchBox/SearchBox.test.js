import React from "react";
import { mount, shallow } from "enzyme";
import SearchBox from "./SearchBox";

describe("SearchBox component", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<SearchBox searchProfile={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call searchProfile function when search button is clicked", () => {
    const mockSearchProfile = jest.fn();
    const wrapper = mount(<SearchBox searchProfile={mockSearchProfile} />);

    const searchInput = wrapper.find("input");
    const searchButton = wrapper.find("button");

    searchInput.simulate("change", { target: { value: "test" } });

    searchButton.simulate("click");

    expect(mockSearchProfile).toHaveBeenCalledWith("test");
  });
});
