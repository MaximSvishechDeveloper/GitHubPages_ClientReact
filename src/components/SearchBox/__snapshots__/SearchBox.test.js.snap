// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component should render without crashing 1`] = `
<div
  className="pa2"
>
  <input
    className="pa3 ba b--green bg-lightest-blue turquoise"
    onChange={[Function]}
    placeholder="Search GitHubPage"
    type="search"
  />
  <button
    className="pa3 ba b--green bg-lightest-blue turquoise"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
